three.js/src/threex/threex-armarkercontrols-nft-end.js

```diff
@@ -0,0 +1,96 @@
+            self.onmessage = function (e) {
+                var msg = e.data;
+                switch (msg.type) {
+                    case "init": {
+                        load(msg);
+                        return;
+                    }
+                    case "process": {
+                        next = msg.imagedata;
+                        process();
+                        return;
+                    }
+                }
+            };
+
+          var next = null;
+
+          var ar = null;
+          var markerResult = null;
+
+          function load(msg) {
+              var camUrl, nftMarkerUrl;
+              var basePath = self.origin;
+              console.log('base path:', basePath);
+              // test if the msg.param (the incoming url) is an http or https path
+              var regex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#()?&//=]*)/igm
+              var re = regex.test(msg.param);
+              console.log(re);
+              if (re == true){
+                camUrl = msg.param;
+              } else if (re == false){
+                camUrl = basePath + '/' + msg.param;
+              }
+              var onLoad = function () {
+                  ar = new ARController(msg.pw, msg.ph, param);
+                  var cameraMatrix = ar.getCameraMatrix();
+
+                  // after the ARController is set up, we load the NFT Marker
+                  if (re == true){
```

Comment:
this will cause issues, because the path is choosen according to the regex for the msg.param, instead i should implement a regex for msg.marker ( the url for the marker!) I'm fixing this!

Reply:
fixed!